resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -59,7 +59,8 @@ private[spark] class BasicExecutorFeatureStep(
     .getOrElse(math.max(
       (kubernetesConf.get(MEMORY_OVERHEAD_FACTOR) * executorMemoryMiB).toInt,
       MEMORY_OVERHEAD_MIN_MIB))
-  private val executorMemoryWithOverhead = executorMemoryMiB + memoryOverheadMiB
+  private val memoryOffHeapMiB = KubernetesUtils.executorOffHeapMemorySizeAsMb(kubernetesConf)
+  private val executorMemoryWithOverhead = executorMemoryMiB + memoryOverheadMiB + memoryOffHeapMiB
```

Comment:
Could you check the other resource manager (YARN/Mesos)? Do we request like this?

Reply:
YARN:
The resource manager for YARN adds the offheap memory to the total memory request. Which is consistent with what one finds in the documentation, in particular the documentation for `spark.executor.memoryOverhead` states:
"The maximum memory size of container to running executor is determined by the sum of `spark.executor.memoryOverhead`, `spark.executor.memory`, `spark.memory.offHeap.size` and `spark.executor.pyspark.memory`. "

Mesos:
I don't work much with Mesos, but from what I can see:
- the documentation for `spark.executor.memoryOverhead` states that "This option is currently supported on YARN and Kubernetes."
- From the code it looks like the memory overhead is implemeted, but just the "standard part", without the off-heap and pyspark memory parts. It should be easy to bring this up to speed with K8S and YARN, in case.

Reply:
yarn does it similar to this. My only other comment might be to reorganize a little to get the pyspark memory before this and then have  executorMemoryWithOverhead = executorMemoryMiB + memoryOverheadMiB + memoryOffHeapMiB + memoryPysparkMiB just for readability.